torch/distributed/_spmd/distribute.py

```diff
@@ -358,24 +351,21 @@ def _get_dtensor_dispatch_graph(
         op_overload = cast(torch._ops.OpOverload, node.target)
 
         if any(a.is_shard() for a in tree_flatten(args)[0] if isinstance(a, DSymInt)):
-            if op_overload in SYM_INT_CONSUMERS:
+            if op_overload in VIEW_SYM_INT_CONSUMERS:
                 assert len(kwargs) == 0, f"Expect empty kwargs, but got {kwargs}"
-                node_to_obj[node] = SYM_INT_CONSUMERS[op_overload](node, args)
-                # skip DTensor expansion
+                node_to_obj[node] = VIEW_SYM_INT_CONSUMERS[op_overload](node, args)
+                return None
             elif op_overload in FACTORY_SYM_INT_CONSUMERS:
                 assert default_mesh is not None, "Requires default mesh for factory ops"
                 node_to_obj[node] = FACTORY_SYM_INT_CONSUMERS[op_overload](
                     node, args, kwargs, default_mesh
                 )
+                return None
             else:
-                # If an operator consumes SymInt sizes on a sharded dimension, we
-                # override with callables in SYM_INT_CONSUMERS or FACTORY_OPS to
-                # create DTensor activations.
-                raise NotImplementedError(
-                    f"{op_overload} consumes SymInt args from a sharded dimension, "
-                    "but SPMD expansion does not support this use case."
+                logger.warn(
```

Comment:
Adding DSymInt support for all ops (e.g., `aten.add.Scalar`, `aten.add.Tensor`, `aten.add.int`, etc.) seems to be unnecessary. It seems to be sufficient to only special-case view and factory ops. All other ops can consume the local_value of DSymInt.